backend/routers/user.py

```diff
@@ -1,27 +1,92 @@
+from pymongo.errors import DuplicateKeyError
+from typing import List
 
-router = APIRouter()
-db = get_database()
+router = APIRouter(
+    prefix="/apps/user",
+    tags=["User"]
+)
+db = getDatabase()
 collection = db["User"]
 
+
 def hash_password(password: str) -> str:
     salt = bcrypt.gensalt()
     hashed = bcrypt.hashpw(password.encode('utf-8'), salt)
     return hashed.decode('utf-8')
 
-@router.post("/addUser", response_model=str)
+
+@router.get("/User")
+async def getAllUserFromDB():
+    users = await collection.find({}, {
+        "_id": 0, 
+        "username": 1,
+        "Email": 1,
+        "first_name": 1,
+        "last_name": 1,
+        "role": 1
+    }).to_list(length=1000)
+    return users
+
+
+
+
+@router.post("/User", response_model=str)
 async def addUser(user: User = Depends(User)):
+    # Hash the password before saving
+    hashed_password = hash_password(user.password)
+    try:
+        await collection.insert_one({"username": user.username, "Email": user.Email, "password": hashed_password, "first_name": user.first_name, "last_name": user.last_name, "apps": user.apps, "role": user.role})
+    except DuplicateKeyError as e:
+
+        if "Email" in str(e):
+            raise HTTPException(
+                status_code=400, detail="Email is already registered")
+        elif "username" in str(e):
+            raise HTTPException(
+                status_code=400, detail="Username is already taken")
+        else:
+            raise HTTPException(
+                status_code=400, detail="A duplicate key error occurred")
+    return "User added successfully!"
+
+
+@router.put("/User", response_model=str)
+async def updateUser(user: updateUser = Depends(updateUser), token=Depends(validateJwtFromCookies)):
+
+    try:
+        token = token.get("Email")
+    except HTTPException as e:
+        return (e.status_code, {"detail": "Authentication failed: Invalid token"})
     # Check if a user with the same username already exists
-    existing_user = await collection.find_one({"username": user.email})
-    if existing_user:
-        raise HTTPException(
-            status_code=400, detail="User with the same username already exists")
 
+    existing_user = await collection.find_one({"Email": token})
     # Hash the password before saving
+    if not existing_user:
+        raise HTTPException(
+            status_code=400, detail="User with the same username does not exist")
     hashed_password = hash_password(user.password)
-    await collection.insert_one({"username": user.username,"Email":user.email, "password": hashed_password,"first_name":user.first_name,"last_name":user.last_name})
-    return "User added successfully!"
\ No newline at end of file
+    await collection.update_one({"Email": token}, {"$set": {"username": user.username, "password": hashed_password, "first_name": user.first_name, "last_name": user.last_name, "apps": user.apps, "role": user.role}})
```

Comment:
This code really needs better formatting. Try using any VSCode extension for formatting.